pkg/errors/errors.go

```diff
@@ -0,0 +1,44 @@
+package errors
+
+import "fmt"
+
+const (
+	// Value for destinguishing non-codedError type.
+	// Not used by codedError.
+	NotCodedErrorType = iota - 1
+
+	UnknownError // Default value
+	ExistError
+	NotValidError
+
+	// Number of error codes. Add new one before it.
+	// Keep last.
+	ErrorMax
+)
+
+type codedError struct {
+	code int
+	text string
+}
+
+func (e codedError) Error() string {
+	return e.text
+}
+
+// Return error code value or NotCodedErrorType if variable isn't of type codedError
+func ErrorCode(e error) int {
```

Comment:
as the package is already named `errors` you can just name it `Code(e error)` or `GetCode(e error)`